pytorch_lightning/trainer/trainer.py

```diff
@@ -805,20 +805,14 @@ def run_predict(self):
         if self.predict_loop.should_skip_predict(max_batches):
             return []
 
-        # ref model
-        model = self.lightning_module
-
-        # enable eval mode + no grads
-        self.predict_loop.on_predict_model_eval()
-        model.zero_grad()
-        torch.set_grad_enabled(False)
+        # call hook
+        self.predict_loop.on_predict_start()
```

Comment:
Why do you call `on_predict_start` before `setup`? It is not done in this order for other hooks (e.g. fit)

Reply:
Because it makes more sense to me to justify the need for 2 hooks.

Reply:
But we should be consistent between hooks otherwise it will bite us down the line

Reply:
+1 to @carmocca 's comment. i'd expect setup to run before on predict start for the same consistency reason.